test/utils/objects/wait/deployment.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wait
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ForDeploymentComplete(cli client.Client, dp *appsv1.Deployment, pollInterval, pollTimeout time.Duration) error {
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		updatedDp := &appsv1.Deployment{}
+		err := cli.Get(context.TODO(), client.ObjectKeyFromObject(dp), updatedDp)
+		if err != nil {
+			// TODO: log
+			return false, err
+		}
+
+		if !IsDeploymentComplete(dp, &updatedDp.Status) {
+			// TODO: log
+			return false, nil
+		}
+
+		return true, nil
+	})
+}
+
+func AreDeploymentReplicasAvailable(newStatus *appsv1.DeploymentStatus, replicas int32) bool {
```

Comment:
None of these functions are waiting, so maybe it could be a little bit confusing when call them using wait package.

Reply:
may be worth moving them back in the parent package, but this will cause (or make much more likely) circular deps. Let's see.